esx_service/vmdk_ops.py

```diff
@@ -844,18 +902,19 @@ def err(string):
     return {u'Error': string}
 
 
-def disk_detach(vmdk_path, vm):
+def disk_detach(vmdk_path, vm, device=None):
```

Comment:
handle_stale_attach() checks for the device to figure if the meta-data is invalid. Like user removed the device (via vim-cmd or VC) from the VM. The meta-data has to be cleared in that case.